internal/types/array.go

```diff
@@ -111,3 +111,38 @@ func (a *Array) Append(values ...any) error {
 func (a *Array) RemoveByPath(path Path) {
 	removeByPath(a, path)
 }
+
+// Contains checks if the Array contains the given value.
+func (a *Array) Contains(value any) bool {
+	for _, v := range a.s {
+		if v == value {
+			return true
+		}
+	}
+	return false
+}
+
+// ContainsAll checks if the Array contains all the given values of the Array b.
+// Currently, this algorithm is O(n^2) without any performance tuning.
+// Possible improvement: to sorts both arrays before comparing.
+// If sorting is implemented, calling of the Contains function will be O(n*log(n)) instead of O(n^2).
+func (a *Array) ContainsAll(b *Array) bool {
+	if a.Len() < b.Len() {
+		// TODO: How to validate that all the values in b are unique?
```

Comment:
> // TODO: How to validate that all the values in b are unique?

Do we need to validate that they are unique? The Array with non-unique array elements is a valid `*Array` type.

Reply:
No, this is not the problem. The problem is this optimization `if a.Len() < b.Len()` doesn't work if `b` has repeated values. Probably, I should just delete the whole optimization to not make this confusion.

Reply:
> Probably, I should just delete the whole optimization to not make this confusion.

If there is optimization, it still works for some arrays.
If you delete optimization, it stops work for that case. 
It doesn't worsen the processing, is it?